### PR TITLE
With the slurm-srun launcher, automatically use prediff-for-slurm-srun.

### PR DESCRIPTION
**Pull request by gbtitus**
We know we need the `prediff-for-slurm-srun` system prediff script to
get clean testing if we're using the slurm-srun launcher, so
`start_test` might as well use it automatically.

### CODE REVIEW

File: util/start_test [hunk 1]
```diff
@@ -1099,6 +1099,12 @@ def set_up_executables():
                         "{0}".format(prediff))
                 finish()
             chpl_system_prediff.append(prediff)
+    if launcher == "slurm-srun":
+        # With Slurm-based launcher, auto-run prediff-for-slurm-srun.
+        pdSlurm = os.path.join(util_dir, "test", "prediff-for-slurm-srun")
+        if not any(pdSlurm in csp for csp in chpl_system_prediff):
```
Comment (ronawho) [inline util/start_test]:
```python
if pdSlurm not in chpl_system_prediff:
```

?

File: util/start_test [hunk 2]
```diff
@@ -1099,6 +1099,12 @@ def set_up_executables():
                         "{0}".format(prediff))
                 finish()
             chpl_system_prediff.append(prediff)
+    if launcher == "slurm-srun":
+        # With Slurm-based launcher, auto-run prediff-for-slurm-srun.
+        pdSlurm = os.path.join(util_dir, "test", "prediff-for-slurm-srun")
```
Comment (ronawho) [inline util/start_test]: `pdSlurm` reads odd to me. I'd probably use `slurm_prediff` or `prediff_for_slurm`

File: util/start_test (hunk 1 above)
Comment (gbtitus) [reply util/start_test]: Oh, I guess I could do that, yeah.  I started out with just `if not any("prediff-for-slurm-srun" in csp ...` and then switched to looking for the entire path.  But when I switched I didn't remove the no-longer-needed `csp for csp in`.

**Comment by ronawho** [comment]
Remove the [`lulesh PREDIFF`](https://github.com/chapel-lang/chapel/blob/master/test/release/examples/benchmarks/lulesh/PREDIFF) since this should eliminate the need for it?

**Comment by gbtitus** [comment]
> Remove the [`lulesh PREDIFF`](https://github.com/chapel-lang/chapel/blob/master/test/release/examples/benchmarks/lulesh/PREDIFF) since this should eliminate the need for it?

Yeah, that's a good idea, but it's not the only one (`library/packages/FFTW` also), I'm going to time out for the week in 16 minutes, and there's not time to test those.  So it's either leave them and do those cleanups later, or change and merge them untested.  Choice?

**Comment by ronawho** [comment]
I'm ok with changing and merging untested -- I will handle any fallout tomorrow

**Comment by ronawho** [comment]
I'm also ok with merging, and removing the prediff later (which may be a smarter move)

**Comment by gbtitus** [comment]
> I'm also ok with merging, and removing the prediff later (which may be a smarter move)

I prefer that route also -- thanks!

**Pull request merged**